internal/cortex/querier/queryrange/instrumentation.go

```diff
@@ -65,3 +96,26 @@ func (c *NoopCollector) Before(ctx context.Context, method string, start time.Ti
 
 // After implements instrument.Collector.
 func (c *NoopCollector) After(ctx context.Context, method, statusCode string, start time.Time) {}
+
+// DurationHistogramCollector collects the duration of a request
+type DurationHistogramCollector struct {
+	metric *prometheus.HistogramVec
+}
+
+func (c *DurationHistogramCollector) Before(ctx context.Context, method string, start time.Time) {
+}
+
+func (c *DurationHistogramCollector) After(ctx context.Context, method, statusCode string, start time.Time) {
+	durationBucket, _ := ctx.Value(QUERYDURATIONBUCKET).(string)
```

Comment:
Check whether the key exists and the type matches. Otherwise, it may panic.
```
durationBucket, ok := ctx.Value(QUERYDURATIONBUCKET).(string)
if !ok {
  durationBucket = "null"
}
```